hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/disk/OnDiskKey.java

```diff
@@ -88,60 +87,42 @@ public K getKey() {
         return key;
     }
 
-    /** Writes the "real" key to the given stream. {@inheritDoc} */
     @Override
-    public void serialize(@NonNull final SerializableDataOutputStream out) throws IOException {
-        writeToStream(out, codec, key);
+    public long getClassId() {
+        // SHOULD NOT ALLOW md TO BE NULL, but ConstructableRegistry has foiled me.
+        return md == null ? CLASS_ID : md.onDiskKeyClassId();
     }
 
     @Override
-    public void serialize(@NonNull final ByteBuffer byteBuffer) throws IOException {
-        serializeReturningWrittenBytes(byteBuffer);
-    }
-
-    public int serializeReturningWrittenBytes(@NonNull ByteBuffer byteBuffer) throws IOException {
-        int initPos = byteBuffer.position();
-        final var output = BufferedData.wrap(byteBuffer);
-        output.skip(Integer.BYTES);
-        codec.write(key, output);
-        final var pos = output.position();
-        output.position(initPos);
-        output.writeInt((int) (pos - initPos - Integer.BYTES));
-        output.position(pos);
-        return (int) (pos - initPos);
+    public int getVersion() {
+        return VERSION;
     }
 
+    /** Writes the "real" key to the given stream. {@inheritDoc} */
     @Override
-    public void deserialize(@NonNull final ByteBuffer byteBuffer, int ignored) throws IOException {
-        final var buff = BufferedData.wrap(byteBuffer);
-        final var len = buff.readInt();
-        final var pos = buff.position();
-        final var oldLimit = buff.limit();
-        buff.limit(pos + len);
-        key = codec.parse(buff);
-        buff.limit(oldLimit);
+    public void serialize(@NonNull final SerializableDataOutputStream out) throws IOException {
+        assert md != null;
```

Comment:
Generally shouldn't use assert in live code, as it's a no-op unless specifically enabled on the command-line.  `Objects.requireNonNull` or a conditional is a better choice here.

Reply:
There is `Object.requireNonNull()` check in OnDiskKey constructor. This assertion is just an extra safety check in debug mode.

I can remove all the checks from `serialize()` and `deserialize()` methods and add a few more to the constructor.